modules/standard/Set.chpl

```diff
@@ -141,7 +150,27 @@ module Set {
       this.parSafe = parSafe;
     }
 
-    // Returns true if the key was added to the hashtable.
+    // Do things the slow/copy way if the element is serializable.
+    // See issue: #17477
+    pragma "no doc"
+    proc _addElem(pragma "no auto destroy" in elem: eltType): bool
```

Comment:
Doesn't this leak?

I am thinking this shouldn't use the pragma, because:

- `elem` is copied into this function, but it will not be freed
- `elem` is also copied on the locale where `this` lives on, and that copy is added to the data structure
- The copy that was created for the function call will leak

I may be missing something, though.

Reply:
It does indeed! I blame `--memLeaks` (though really, myself). Removed `pragma "no auto destroy"`, thanks!